src/module/entities/helper.js

```diff
@@ -6,97 +6,1302 @@
+
+				if (!entityDocuments.length > 0) {
+					return resolve({
+						message: constants.apiResponses.ENTITY_NOT_FOUND,
+						result: [],
+					})
+				}
+
+				let result = []
+
+				//  if( rolesDocument[0].entityTypes[0].entityType === constants.common.STATE_ENTITY_TYPE ) {
+				//     result = entityDocuments[0].childHierarchyPath;
+				//     result.unshift(constants.common.STATE_ENTITY_TYPE);
+				//  } else {
+
+				//     let targetedEntityType = "";
+
+				//     rolesDocument[0].entityTypes.forEach(singleEntityType => {
+				//        if( entityDocuments[0].childHierarchyPath.includes(singleEntityType.entityType) ) {
+				//            targetedEntityType = singleEntityType.entityType;
+				//        }
+				//     });
+
+				// let findTargetedEntityIndex =
+				// entityDocuments[0].childHierarchyPath.findIndex(element => element === targetedEntityType);
+
+				// if( findTargetedEntityIndex < 0 ) {
+				//    throw {
+				//        message : CONSTANTS.apiResponses.SUB_ENTITY_NOT_FOUND,
+				//        result : []
+				//    }
+				// }
+
+				// result = entityDocuments[0].childHierarchyPath.slice(findTargetedEntityIndex);
+
+				return resolve({
+					success: true,
+					message: CONSTANTS.apiResponses.ENTITIES_CHILD_HIERACHY_PATH,
+					result: entityDocuments,
+				})
+			} catch (error) {
+				return reject(error)
+			}
+		})
+	}
+
+	/**
+	 * update registry in entities.
+	 * @method
+	 * @name listByLocationIds
+	 * @param {Object} locationIds - locationIds
+	 * @returns {Object} entity Document
+	 */
+
+	static listByLocationIds(locationIds) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// Constructing the filter query to find entities based on locationIds
+				let filterQuery = {
+					$or: [
+						{
+							'registryDetails.code': { $in: locationIds },
+						},
+						{
+							'registryDetails.locationId': { $in: locationIds },
+						},
+					],
+				}
+
+				// Retrieving entities that match the filter query
+				let entities = await entitiesQueries.entityDocuments(filterQuery, [
+					'metaInformation',
+					'entityType',
+					'entityTypeId',
+					'registryDetails',
+				])
+				if (!entities.length > 0) {
+					throw {
+						message: CONSTANTS.apiResponses.ENTITIES_FETCHED,
+					}
+				}
+
+				return resolve({
+					success: true,
+					message: CONSTANTS.apiResponses.ENTITY_FETCHED,
+					data: entities,
+				})
+			} catch (error) {
+				return resolve({
+					success: false,
+					message: error.message,
+				})
+			}
+		})
+	}
+
+	/**
+	 * find detils in entities.
+	 * @method
+	 * @name find
+	 * @param {Object} bodyQuery - body data
+	 * @param {Object} projection - projection to filter data
+	 */
+
+	static find(bodyQuery, projection) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// Fetch entities based on the provided query and projection
+				const result = await entitiesQueries.entityDocuments(bodyQuery, projection)
+				return resolve({
```

Comment:
@MallanagoudaB 
All these logic you removed not required here?

if (entityDocument.length < 1) {
					throw {
						status: HTTP_STATUS_CODE.not_found.status,
						message: CONSTANTS.apiResponses.ENTITY_NOT_FOUND,
					}
				}

				let relatedEntities = await entitiesHelper.relatedEntities(
					entityDocument[0]._id,
					entityDocument[0].entityTypeId,
					entityDocument[0].entityType,
					projection
				)
				console.log(relatedEntities, 'line no 104')
				_.merge(result, entityDocument[0])
				let relatedEntities = await entitiesHelper.relatedEntities(req.params._id)
				result['relatedEntities'] = relatedEntities.length > 0 ? relatedEntities : []

Reply:
@MallanagoudaB This old logic is not required?

Reply:
we dont require that logic because i changed whole logic because its already there in the helper so im not calling that helper function